src/ti_comm/src/new_lib/amp_serial_jetson_SERIAL_TEST.cpp

```diff
@@ -49,6 +52,36 @@ FILE * fptr3 = fopen("debug_rx.txt", "w");
 #endif
 
 int main(int argc, char** argv) {
+    // Command pair structured as: <speed, angle>
+    queue<pair<float, float> > commands;
+    // File to read commands from 
+    string filename = "debug.txt";
+    // Default delay value of 0.05
+    float delay = 0.05;
+    string currSpeed;
+    string currAngle;
+
+    argparse(argc, argv, filename, delay);
+
+    // Handle invalid/corrupt debug file
+    ifstream fin(filename);
+    if (fin.fail()) {
+        cout << "Error: File " << filename << " doesn't exist or is incorrectly formatted" << endl; 
+        return 1;
+    }
+
+    // Read commands from file
+    string currSpeed;
+    string currAngle;
+    while (fin) {
+        getline(fin, currSpeed, ' ');
+        getline(fin, currAngle);
+        if (currAngle.empty() || currSpeed.empty()) break;
+        float speed = stof(currSpeed);
+        float angle = stof(currAngle);
+        commands.push(make_pair(speed, angle));
+    }
```

Comment:
```suggestion
    float currSpeed;
    float currAngle;
    while (fin >> currSpeed && fin >> currAngle) {
        commands.push(make_pair(speed, angle));
    }
```
It has been a minute since I've done C++ but I think this should work. Feel free to disregard though as this is a minor suggestion (usually called a "nit").

Reply:
Yeah I've tested this out, looks good to me.